includes/class-wsuwp-people-directory-page-template.php

```diff
@@ -446,23 +449,40 @@ public function save_post( $post_id ) {
 			}
 		}
 
-		// Set a flag to flush rewrite rules.
-		if ( false === apply_filters( 'wsuwp_people_default_rewrite_slug', false ) ) {
-			set_transient( 'wsuwp_people_directory_flush_rewrites', true );
-		}
-
 		// Update associated people data.
 		if ( ! isset( $_POST['_wsu_people_directory_profile_ids'] ) ) {
 			return;
 		}
 
 		// Stop here if the order of people hasn't changed.
-		//if ( $previous_ids === $_POST['_wsu_people_directory_profile_ids'] ) {
-		//	return;
-		//}
+		if ( $previous_ids === $_POST['_wsu_people_directory_profile_ids'] ) {
+			return;
+		}
 
-		// Save order data of the associated people.
-		$ids = explode( ' ', $_POST['_wsu_people_directory_profile_ids'] );
+		// Set a flag to flush rewrite rules.
+		if ( false === apply_filters( 'wsuwp_people_default_rewrite_slug', false ) ) {
+			set_transient( 'wsuwp_people_directory_flush_rewrites', true );
+		}
+
+		$scheduled_event_args = array(
+			$_POST['_wsu_people_directory_profile_ids'],
+			$post_id,
+		);
+
+		wp_schedule_single_event( time(), 'wsuwp_people_add_or_update_profiles', $scheduled_event_args );
+		wp_schedule_single_event( time(), 'wsuwp_people_remove_profile_from_page', $scheduled_event_args );
+	}
+
+	/**
+	 * Add or update profiles associated with a directory page.
+	 *
+	 * @since 0.3.5
+	 *
+	 * @param array $ids     The IDs for the primary profile records.
+	 * @param int   $post_id The directory page ID.
+	 */
+	public function add_or_update_profiles( $profile_ids, $post_id ) {
+		$ids = explode( ' ', $profile_ids );
```

Comment:
It may also be that the sanitization for the previous comment should occur here instead.